cpp/src/arrow/compute/key_hash_avx2.cc

```diff
@@ -190,7 +190,7 @@ uint32_t Hashing32::HashFixedLenImp_avx2(uint32_t num_rows, uint64_t length,
   // Do not process rows that could read past the end of the buffer using 16
   // byte loads. Round down number of rows to process to multiple of 2.
   //
-  uint64_t num_rows_to_skip = bit_util::CeilDiv(length, kStripeSize);
+  uint64_t num_rows_to_skip = bit_util::CeilDiv(kStripeSize, length);
```

Comment:
Is this right? I don't understand why we would divide a constant-time stripe size by a runtime length.
Don't you instead simply want something like:
```c++
  uint32_t num_rows_to_process = length / kStripeSize * kStripeSize;
```

Reply:
I guess you were thinking `length` is some kind of "total length" of all rows? But it actually represents the width of the fixed length type being hashed. See its call site:
https://github.com/apache/arrow/blob/9c0a7617d953a74abaec5c14318bc2f46d0d176a/cpp/src/arrow/compute/key_hash.cc#L431C69-L431C78

Reply:
Oh, darn. Could we take the opportunity to update signature and call sites? Call it `row_length` for example.

Reply:
I guess we better do so. Let me update them in this PR.

Reply:
Just to confirm before I make the change, how do you think the name `fixed_length`? I think it might be more specific.

Reply:
Or `key_width`?

Reply:
> Actually, perhaps `key_length`?

Deal.

Reply:
Done.

Changed every misleading `length` to `key_length`. Additionally changed some other names to make naming slightly more consistent between header and cpp, function counterparts.